Note update to NoClassDefFoundError exception message

sharon-wang opened the pull request:
Update What's New for 0.21 to include `NoClassDefFoundError` exception message update

Fixes: https://github.com/eclipse/openj9-docs/issues/563
Related: https://github.com/eclipse/openj9/pull/9588

sharon-wang commented:
@pshipton Should more detail be provided in this note?

pshipton commented:
We can include an example, and describe the meaning of each name in the message.

sharon-wang commented:
@pshipton I added a simple example and noted that one class name is the "name encountered by the VM" and the other class name is the "wrong name provided to the VM". I can try to reword the description if it's a bit confusing.

pshipton commented on docs/version0.21.md:
```diff
@@ -48,6 +49,30 @@ To learn more about support for OpenJ9 releases, including OpenJDK levels and pl
+root@mycomputer:~/path/to/mypackage javac Main.java
+root@mycomputer:~/path/to/mypackage java Main
+Error: Could not find or load main class Main
+Caused by: java.lang.NoClassDefFoundError: mypackage/Main (wrong name: Main)
```
By an example, I just meant this line and the following paragraph. I don't think the code example to recreate it is necessary, but it doesn't need to be removed unless the docs team thinks it's too much for a What's New.

pshipton commented on docs/version0.21.md:
````diff
@@ -48,6 +49,30 @@ To learn more about support for OpenJ9 releases, including OpenJDK levels and pl
+Error: Could not find or load main class Main
+Caused by: java.lang.NoClassDefFoundError: mypackage/Main (wrong name: Main)
+```
+Where `mypackage/Main` is the class name encountered by the VM, but "wrong name" `Main` was the provided class name. Prior to this update to the exception message, the encountered class name and the provided class name were swapped in the `NoClassDefFoundError` exception message.
````
could this say something like `encountered by the VM in the .class file`.

sharon-wang commented:
@pshipton I moved the long example to my openj9 [PR](https://github.com/eclipse/openj9/pull/9588) description and updated the explanation to include `encountered by the VM in the .class file`.

sharon-wang commented:
@pmhayward Hi Peter, this is ready for review.

pmhayward commented:
Jenkins doc stage

pmhayward commented:
lgtm... thanks!
